istioctl/pkg/multicluster/remote_secret.go

```diff
@@ -389,21 +390,16 @@ func createRemoteSecret(opt RemoteSecretOptions, client kubernetes.Interface, en
 		opt.ClusterName = string(uid)
 	}
 
-	tokenSecret, err := getServiceAccountSecretToken(client, opt.ServiceAccountName, opt.Namespace)
-	if err != nil {
-		return nil, fmt.Errorf("could not get access token to read resources from local kube-apiserver: %v", err)
-	}
-
-	server, err := getServerFromKubeconfig(opt.Context, env.GetConfig())
-	if err != nil {
-		return nil, err
-	}
 	var secretName string
+	var err error
 	switch opt.Type {
 	case SecretTypeRemote:
 		secretName = remoteSecretNameFromClusterName(opt.ClusterName)
 	case SecretTypeConfig:
 		secretName = configSecretName
+		if opt.ServiceAccountName == DefaultServiceAccountName {
+			opt.ServiceAccountName = DefaultConfigServiceAccountName
```

Comment:
This is unexpected behavior. We need to make sure we document it in the flag description.

Reply:
add description into flag:

```
      --service-account --type              Create a secret with this service account's credentials. Use "istio-reader-service-account" as default value if --type is "remote", use "istiod-service-account" as default value if --type is "config"
```